### PR TITLE
Enable composefs for 41+

### PR DESCRIPTION
**Pull request by cgwalters**
We really want to aim to eventually enable this by default, let's test things out in rawhide.

A thing that this is known to break is the "chattr -i" hack for new toplevel dirs (xref https://github.com/coreos/rpm-ostree/issues/337 )

Basically if you want that, you either need to make a derived image, or enable transient root.

### CODE REVIEW

**Comment by cgwalters** [comment]
I verified this didn't enable composefs for testing-devel, but I didn't verify that cherry picking this to rawhide does enable it.

At the moment, this blocks on https://bodhi.fedoraproject.org/updates/FEDORA-2024-555d4e8de6 getting into coreos-assembler.

File: manifests/fedora-coreos.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ conditional-include:
     # Remove this once we have Podman v5 in f39.
     # xref: https://github.com/coreos/fedora-coreos-tracker/issues/1629
     include: cni-plugins.yaml
+  - if:
+      - releasever >= 41
```
Comment (dustymabe) [inline manifests/fedora-coreos.yaml]:
```suggestion
  - if: stream == "rawhide"
```

Let's actually match against `rawhide` here. Eventually F41 will come to a prod stream. If we choose to turn composefs on for prod streams we should open a ticket for it and record the results of the discussion and then make the change.

**Comment by dustymabe** [comment]
To give this a full shake I kicked off a run in staging: https://jenkins-fedora-coreos-pipeline.apps.ocp.stg.fedoraproject.org/blue/organizations/jenkins/build/detail/build/115/pipeline

**Comment by dustymabe** [comment]
> To give this a full shake I kicked off a run in staging: https://jenkins-fedora-coreos-pipeline.apps.ocp.stg.fedoraproject.org/blue/organizations/jenkins/build/detail/build/115/pipeline

In that run I did a replay of a previous one where I cut out kola tests.. Here's a new run: https://jenkins-fedora-coreos-pipeline.apps.ocp.stg.fedoraproject.org/blue/organizations/jenkins/build/detail/build/116/pipeline

**Comment by cgwalters** [comment]
> A thing that this is known to break is the "chattr -i" hack for new toplevel dirs (xref https://github.com/coreos/rpm-ostree/issues/337 )

What I don't know offhand is how problematic that will be in reality.  My instinct says we can do this, but with a sufficiently loud announcement.

**Comment by cgwalters** [comment]
podman-machine before 5.0 was doing the chattr, but I believe it no longer is - need to do check.

**Comment by jbtrystram** [comment]
I wanted to experiment with that so I rebased this patch and built a rawhide with it and it works as expected : 

```
[core@cosa-devsh ~]$ findmnt /
TARGET SOURCE  FSTYPE  OPTIONS
/      overlay overlay ro,relatime,seclabel,lowerdir=/run/ostree/.private/cfsroot-lower::/sysroot/ostree/repo/objects,redirect_dir=on,metacopy=on

[root@cosa-devsh core]# ls -la /sysroot/ostree/boot.1/fedora-coreos/ae2957172d56a1da0d18a974fcc50b515d08ec0a6c231a1ef78975130e1228d6/0/
total 5652
drwxr-xr-x. 12 root root    4096 May 28 15:28 .
drwxr-xr-x.  3 root root     161 Aug  1  2022 ..
-rw-r--r--.  1 root root 5771264 Aug  1  2022 .ostree.cfs
```

**Comment by jlebon** [comment]
Chatted with @jbtrystram about this. We can open a PR against the rawhide branch with this change just so CI can run against it properly. If all the issues have been ironed out, we can get this in.

**Comment by jbtrystram** [comment]
Closing this for #3009

**Pull request closed without merging**